email_alert_service/listeners/listener.rb

```diff
@@ -5,7 +5,7 @@ def initialize(queue_binding, processor)
   end
 
   def listen
-    @queue_binding.subscribe(block: true, manual_ack: true) do |delivery_info, properties, document_json|
+    @queue_binding.subscribe(block: false, manual_ack: true) do |delivery_info, properties, document_json|
```

Comment:
Interesting... just reading here [1] about the block option. It says: "If blocking the calling thread is really necessary, pass :block => true to Bunny::Queue#subscribe. Note that this may affect topology recovery and is not recommended for production code"

1: http://rubybunny.info/articles/queues.html

Reply:
This is a good point. I'm still using a similar blocking method in this code and now I'm wondering if I should not use the sleep method as in your article.

Reply:
I changed the code in line with the suggestion from that article.

Reply:
good find by they way 👍